src/HTTPClient.ts

```diff
@@ -27,10 +27,41 @@ export class HTTPClient {
 	 * @public
 	 * @constructor
 	 * Creates an instance of a HTTPClient
-	 * @param {Middleware} middleware - The first middleware of the middleware chain
+	 * @param {...Middleware} middleware - The first middleware of the middleware chain or a sequence of all the Middleware handlers
 	 */
-	public constructor(middleware: Middleware) {
-		this.middleware = middleware;
+	public constructor(...middleware: Middleware[]) {
+		this.setMiddleware(...middleware);
+	}
+
+	/**
+	 * @private
+	 * Processes the middleware parameter passed to set this.middleware property
+	 * @param {...Middleware} middleware - The middleware passed
+	 * @returns Nothing
+	 */
+
+	private setMiddleware(...middleware: Middleware[]): void {
+		if (middleware.length > 1) {
+			this.parseMiddleWareArray(middleware);
+		} else {
+			this.middleware = middleware[0];
```

Comment:
This line will throw if the array is empty.

Reply:
I did consider the case if the middleware is null or empty went with the assumption that if the user is not inputting a valid middleware sequence we shouldn't be letting the user continue and throw an error.

Reply:
I checked with Typescript playground, if no parameter is provided to a varargs param, the indexing will resolve into undefined (must be a very good reason why they did that).

So it will be silent error, which will surface at another place. I think we should be more explicit here with this edge case.

Side note: even if it was throwing at this line, an explicit error message would have been better instead of throwing in the first line that tries to index into the array.

Reply:
I added the check on the middleware parameter passed.